src/linear_algebra/basis.lean

```diff
@@ -758,6 +758,12 @@ hv.1.total_repr ⟨x, _⟩
 lemma is_basis.total_comp_repr : (finsupp.total ι M R v).comp hv.repr = linear_map.id :=
 linear_map.ext hv.total_repr
 
+lemma is_basis.ext {f g : M →ₗ[R] M'} (hv : is_basis R v) (h : ∀i, f (v i) = g (v i)) : f = g :=
```

Comment:
generalize `hv` to that the image spans?

Reply:
Sounds good. I'll make a new, more general lemma `linear_map.ext_on` and keep `is_basis.ext` since it was already used before this PR.